### ISSUE TITLE
ask for which From: Adress to use

### ISSUE THREAD

**Issue by toogley** [neomutt/neomutt#276]
Hey.

I use multiple accounts in mutt and sometimes, i forget to check which account I use for sending mails.

Currently, when I compose a new message, the account is selected based on the `From: ` entry in the Mail. Mutt already asks for the Subject and the recipients - I would very happy If mutt would ask me about the `From: ` entry too.

What do you think about that?

**Comment by guyzmo** [comment]
you've got several ways to work it out, first you can modify that when you're editing your mail:

```
set edit_headers		# let me edit the message header when composing
```

then you can make a few aliases to make it faster to select your from:

```
alias me_home me@home.example.org
alias me_work me@work.example.org
```

then you can set the `alternates` to select what From you want to auto fill based on the incoming mail (and the others that you won't match, you just leave them blank)

```
alternates me.*@.*\.example.org
```

then you, on the compose summary view, when you set:

```
alias identity_1 Me Work <me@work.example.org>
alias identity_2 Me Home <me@home.example.org>

macro compose v "<edit-from>^Uidentity_<tab>" "Select from"
```

the for the first two if you press `meta-f` (or `esc` followed by `f`), and then `tab` you'll get the list of aliases. The macro `v` is a shorter version to directly choose the from from the list.

Basically, there are many ways mutt already handles that ☺

**Comment by toogley** [comment]
Hey :)

Basically, i already do most of what you've described. My problem is: I am very forgetful - so it is very common that i forget to control the From address and send emails under the wrong identify. If I'm prompted for the correct one, it would make my email management much easier.

**Comment by guyzmo** [comment]
well, I guess that could be done, but it's quite redundant with having it in your editor *and* in the compose summary view…

Or maybe a good idea would be to go straight to the compose summary view before editing? Because actually my big mistake with vim is forgetting to select attachments before sending ☺

**Comment by flatcap** [comment]
> my big mistake with vim is forgetting to select attachments before sending

Sounds like you'd benefit from the [forgotten attachment](https://www.neomutt.org/feature/forgotten-attachment) feature :-)

**Comment by guyzmo** [comment]
I have it on, but still finding new ways to miss those ☺

Though I think having the compose screen first and no questions could be a decent way to prepare mails. So you can select the From, add the attachments, eventually the To and/or the CCs… And then you can write the mail! ☺ Having the compose screen first helps set the mind up before writing the mail, whereas when I write a (lengthy and hard to write) mail, I want to get over with it and then I do `ZZy` to exit vim and send mail and then I say *damn… I forgot X*.

I if you do `set editor=touch %s` you get straight to the compose summary screen, and then you reset `set editor=vim` and you're set. That's a very ugly and stupid hack. Maybe a simple config option would be nicer?

**Comment by flatcap** [comment]
@guyzmo 
`set edit_headers = yes` also allows you a little trick that might help you.
If you add `attach: myfile.txt` to the header section, the file's automatically attached.
You can tweak most (?) of the headers from within vim.

**Comment by guyzmo** [comment]
>  If you add attach: myfile.txt to the header section, the file's automatically attached.

what the. really. ugh. O_O

I actually imagend that having `attach:` as a header would be a great thing to have, but never hoped for it to be a real thing.

ok scratch that. I personally don't care for the summary before or after or whatever. Vim is all. Vim is ♥. And the attach header is its prophet!

**Comment by guyzmo** [comment]
so, @toogley, basically either you set up your from in the editor, or in the summary page, but don't count much on the questions before composing your mail.

Actually once we insert scripting capabilities to neomutt, you'll be able to make your own extension to handle that. So I'd vote for closing this issue as "won't do", and stay tuned for the scripting extension to work it out ☺

**Comment by protist** [comment]
@toogley Another option would be to use [mutt-vid](https://github.com/protist/mutt-vid/) (I'm the dev). This basically saves the last sender that you used to send an email to recipient Foo. The next time you send an email again to Foo, it will remember this sender you last used, and use it again.

**Comment by edison23** [comment]
Hi guys,
after reading this discussion, maybe I'm missing something, but is there a way to make mutt use (as a sender) the address the mail came to? For me, it always uses the default one (or maybe it's just a bad config on my side, dunno).

**Comment by protist** [comment]
@edison23 You mean when directly replying to an email? `reverse_name` [does this](http://www.mutt.org/doc/manual/#reverse-name).

**Comment by edison23** [comment]
@protist oh... kinda missed that one..  it works! thanks :)
